APIScrape/scrape.php

```diff
@@ -0,0 +1,176 @@
+}
+
+/**
+ * Puts the list of breeds into the database.
+ * 
+ * @param PDO $db - the database to insert into.
+ * @param array $catBreeds - the list of cat breeds. As breed id => breed name.
+ * @return void
+ */
+function fillCatBreedToDB(PDO $db, array $catBreeds)
+{
+    foreach($catBreeds as $id => $breed) {
+        $sql = $db->prepare('INSERT INTO `breed` (breed) VALUES (:breedToString);');
+        $breedToString = "$breed";
+        $sql->bindParam('breedToString', $breedToString, PDO::PARAM_STR);
+        $sql->execute();
+    }
+    echo "Cat breeds added to database.\n";
+}
+
+/**
+ * Use the list of cat breeds to make the API requests so we can get URLs
+ * of cat pictures to link to in the app.
+ * 
+ * @param array of cat breeds
+ * @return array of breed names => array of strings representing the image URLs
+ */
+function getCatImageURLs(array $catBreeds):array
+{
+    $imageSourceArray = [];
+    $breedCount = count($catBreeds);
+    $i = 1;
+    echo "Fetching cat image URLs...\n";
+    foreach ($catBreeds as $id => $name) {
+        $catImageApiUrl = 'https://api.thecatapi.com/v1/images/search?breed_ids=' . $id . '&limit=21';
+        $curl = curl_init();
+        curl_setopt_array($curl, array(
+            CURLOPT_URL => $catImageApiUrl,
+            CURLOPT_CUSTOMREQUEST => "GET",
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_HTTPHEADER => [ API_KEY ]
+        ));
+        $imageApiResponse = curl_exec($curl);
+        if($imageApiResponse === false) {
+            echo 'Curl error: ' . curl_error($curl);
+        } else {
+            // Little bash technique to clean up console reporting
+            // Use bash escape \033[2K to clear the line before writing the message
+            // \r carriage returns to the beginning of the line before writing the next
+            // The console lines write in-place - much neater
+            echo "\033[2KReceived cat image URLs for breed '$name' successfully ($i of $breedCount).\r";
+        }
+        $i++;
+
+        curl_close($curl);
+
+        $responseArray = json_decode($imageApiResponse, true);
+
+        $breedImages = [];
+        foreach($responseArray as $item) {
+            array_push($breedImages, $item["url"]);
+        }
+        $imageSourceArray[$name] = $breedImages;
+    }
+    return $imageSourceArray;
+}
+
+/**
+ * Puts the cat image URLs into the database.
+ * 
+ * @param PDO $db the database to use
+ * @param array The list of cat breeds
+ * @param array The list of cat image URLs
+ * @return void
+ */
+function fillCatImages(PDO $db, array $catBreedArray, array $catImageSourceArray)
+{
+    $catBreedIndexedArray = [];
+    foreach($catBreedArray as $breed) {
+        $catBreedIndexedArray[] = $breed;
+    }
+    $sqlArray = [];
+    for($breedIndex = 0; $breedIndex < count($catImageSourceArray); $breedIndex++) {
+        foreach($catImageSourceArray[$catBreedIndexedArray[$breedIndex]] as $url) {
+            $breedID = $breedIndex + 1;
```

Comment:
Could you add a comment explaining that this value matches the breed id in the table?

Reply:
on it